### ISSUE TITLE
Roster Error when Removing User: Missing attribute 'id' in tag <iq/> 

### ISSUE THREAD

**Issue by eduncan911** [boothj5/profanity#946]
Update: version numbers below.

Profanity: 0.4.4 (whonix version)
libstrophe: (doesn't seem to be installed, at least not via aptitude)

---

**What are you trying to do:**

Remove a user from the Roster for a domain that no longer exists using `/roster remove user@non-existant-domain.com`.  

Note that Profanity displays an error _on startup_ that these users in my roster had an error: domain not found.  I think this is part of the issue, as it cannot create an `id` per say.

**Expected Result:**

When issunig `/roster remove user@domain.com`, it should remove the user from MY account on MY server - regardless if the remote domain exists or not.

**What actually happens:**

Now that `dukgo.com`'s xmpp has gone the way of the Dodo bird (it's been decommissioned and turned off), I had several users on dukgo that show up in my Roster.  I am trying to remove them.

When I try `/roster remove user@domain.com`, nothing happens. 

Enabling DEBUG log, I see this:

```
24/06/2017 16:38:23: prof: DBG: Input received: /roster remove user@domain.com
24/06/2017 16:38:23: conn: DBG: SENT: <iq type="set"><query xmlns="jabber:iq:roster"><item jid="user@domain.com" subscription="remove"/></query></iq>
24/06/2017 16:38:23: xmpp: DBG: RECV: <iq type="error"><query xmlns="jabber:iq:roster"><item jid="user@domain.com" subscription="remove"/></query>
<error code="400" type="modify"><bad-request xmlns="urn:ietf:params:xml:ns:xmpp-stanzas"/><text xmlns="urn:ietf:params:xml:ns:xmpp-stanzas">Missing attribute 'id' in tag &lt;iq/&gt; qualified by namespace 'jabber:client'</text></error></iq>
24/06/2017 16:38:23: prof: ERR: IQ error received.
```

And the users continue to be listed on the server, and therefore locally.  

Profanity continues to give me an error on startup that user@dukgo.com cannot be found - and for each reconnect, it floods me with these for each other.

**Reproduction Steps**

Well, you'll somehow need to add a Valid user to your XMPP account first.  And then somehow remove that xmpp server from working remotely - thereby forcing Profanity to now display an error that the domain doesn't exist on startup.

You could possibly add a real Temp user, shutdown Profanity and then change your Hosts file to point that xmpp server to a 127.0.0.1 location.  Then startup Profanity and it should generate that error that the domain cannot be found.  

Then try to remove that user from the roster.

---

More info:

If I use another Jabber client, like Thunderbird AND Pidgin, I can remove the user without issue.  I actually tried both and they were able to remove `dukgo.com` users from my roster.  And when logging into Profanity, they are gone!

I left two there for now, to help with debugging/validating any fix to this error.  I might remove them at a later time if it just annoys me too much.

**Comment by pasis** [comment]
Could you also say what is you profanity and libstrophe version (commit if you built it from repository). I found it strange that `id` attribute is missed. All I can see that could lead to the error are memory allocation error or some error in libstrophe.

**Comment by eduncan911** [comment]
Updated with version number (0.4.4).

I now see this is an old version.  I will try to pull latest and compile it when I have some time.  Thank you!

**Comment by pasis** [comment]
I checked profanity-0.4.4 code and there is error there:
```
xmpp_stanza_t *
stanza_create_roster_remove_set(xmpp_ctx_t *ctx, const char * const barejid)
{
    xmpp_stanza_t *iq = xmpp_stanza_new(ctx);
    xmpp_stanza_set_name(iq, STANZA_NAME_IQ);
    xmpp_stanza_set_type(iq, STANZA_TYPE_SET);

    xmpp_stanza_t *query = xmpp_stanza_new(ctx);
    xmpp_stanza_set_name(query, STANZA_NAME_QUERY);
    xmpp_stanza_set_ns(query, XMPP_NS_ROSTER);

    xmpp_stanza_t *item = xmpp_stanza_new(ctx);
    xmpp_stanza_set_name(item, STANZA_NAME_ITEM);
    xmpp_stanza_set_attribute(item, STANZA_ATTR_JID, barejid);
    xmpp_stanza_set_attribute(item, STANZA_ATTR_SUBSCRIPTION, "remove");

    xmpp_stanza_add_child(query, item);
    xmpp_stanza_release(item);
    xmpp_stanza_add_child(iq, query);
    xmpp_stanza_release(query);

    return iq;

}
```
As you see, no `id` attribute.

**Comment by eduncan911** [comment]
So.. updating to latest should resolve the issue?

**Comment by pasis** [comment]
Yes, 0.5.x has correct code. Maybe it was fixed even earlier, but 0.5.1 should be in most of popular distros.

**Comment by eduncan911** [comment]
Confirmed, downloading latest tarbell and building resolved it.

Whonix is based on debian jessie, and Jessie currently has 0.4.4 as current package.
